src/v/cloud_storage/tests/remote_test.cc

```diff
@@ -645,6 +645,111 @@ FIXTURE_TEST(test_list_bucket, remote_fixture) {
     }
 }
 
+FIXTURE_TEST(test_list_bucket_with_max_keys, remote_fixture) {
```

Comment:
I'm not sure it's worth checking in, but it's worth at least trying to run this test to make sure this passes https://gist.github.com/andrwng/941bf4eca34a020aac59c390ddb9a4c3

Reply:
Tested, and confirmed that this passes. `test_list_bucket_with_max_keys` tests a similar case in which the `s3_imposter`'s "bucket" is filled with more than the default `max_keys` when we attempt to list from it. Although, it's nice to have confirmation on the MinIO side as well.